hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector_purgeTest.java

```diff
@@ -73,7 +73,12 @@ public void testPurging_Invocation() {
         Collection<SlowOperationLog> logs = getSlowOperationLogsAndAssertNumberOfSlowOperationLogs(instance, 1);
 
         SlowOperationLog firstLog = logs.iterator().next();
-        assertTotalInvocations(firstLog, 4);
+        int totalInvocations = firstLog.totalInvocations.get();
+
+        //If due to race condition unable to collect a stacktrace of the operation before the next operation so the invocation
+        //will be dropped.
+        assertTrue(String.format("Expected 3 or 4 total invocations, but was %s, logs: %s", totalInvocations,
+                firstLog.createDTO().toJson()), totalInvocations >= 3 && totalInvocations <= 4);
```

Comment:
As I understand, this test is supposed to verify that we purge captured invocations while we keep the `SlowOperationLog` instance itself, updating its total invocation counter. So I think we need to verify only that if we purged anything by simply checking if `totalInvocations > invocations.size()`. Also, we can execute a few "quicker" EPs, sleep a longer to ensure there is purging and then we can verify. The problem is that occasionally we have `>2s` stalls in the PR builder :) WDYT?

Reply:
Indeed, checking the fact of purging instead of exact numbers is a good idea. It will make the test more specific. Thank you for the advice.

Making EPs quicker is not recommended here because the checking thread works every 1 sec, and in first run it only registers operation start time. So the first operation can be marked as slow only after 2s of actual execution. 

For more probable purging I increased one ER for 1 sec.